### PR TITLE
Remove unused cruft from clr

### PR DESCRIPTION
**Pull request by mpcusack-color**
Mostly to support global injectable middleware that hasn't actually been in use since before this repo was created.

Next up: https://github.com/color/clr/pull/14

### CODE REVIEW

File: clr/config.py [hunk 1]
```diff
@@ -23,16 +24,5 @@ def find_clrfile(name='clrfile.py'):
...
-    config = _get_config()
-    return config['options']
-
-_config_cache = {}
-def _get_config():
-    if not _config_cache:
-        execfile(find_clrfile(), _config_cache)
-    return _config_cache
+def read_namespaces():
+    return runpy.run_path(find_clrfile())['commands']
```
Comment (DavidCain) [inline clr/config.py:28]: I've not used `runpy` before (looks neat!) - can you supply the rationale for using it here? (A quick method docstring would probably be helpful to other readers too)

Comment (mpcusack-color) [reply clr/config.py:39]: First time using it as well. execfile is deprecated and this is in the stdlib.

File: clr/config.py [hunk 2]
```diff
@@ -2,6 +2,11 @@
 from pathlib import Path
 import os.path
 import os
+import runpy
+
+"""The clr tool is configred using a python source file (clrfile.py) that is
```
Comment (DavidCain) [inline clr/config.py:7]: Moving this docstring above the imports will make it the value of `__doc__` for the module (where it is now, it's just an unused string literal).

File: clr/commands.py [hunk 3]
```diff
@@ -129,6 +131,7 @@ class Namespace:
     longdescr: str
     command_specs: Dict[str, CommandSpec]
     command_callables: Dict[str, Callable]
+    instance: any
```
Comment (DavidCain) [inline clr/commands.py:134]: `any` is a method (which takes an iterable, returning a boolean if any elements are true)

You probably want `typing.Any`.

File: clr/config.py (hunk 2 above)
Comment (josiah-co) [reply clr/config.py:7]: Can also be explicit by assigning via `__doc__ = `.

Comment (DavidCain) [reply clr/config.py:7]: Yup, that definitely does work, but [docstring convention](https://www.python.org/dev/peps/pep-0257/) is to just put the string literal at the top of the file:

> A docstring is a string literal that occurs as the first statement in a module, function, class, or method definition. Such a docstring becomes the `__doc__` special attribute of that object.

In general, I think it's best to just let the built-in parts of the language handle interacting with dunder values where possible. (e.g. `a.__add__(b)` works, but one can also just do `a + b`).

**Review by josiah-co — APPROVED** [review]
These deletions make me so happy.

File: clr/commands.py [hunk 4]
```diff
@@ -306,3 +309,7 @@ def arg_text(arg_name, default_value):
         if docstr:
             for line in docstr.split('\n'):
                 print(text_wrapper.fill(line))
+
+    def cmd_argtest(self, a, b, c=4, d=None, e=False):
```
Comment (DavidCain) [inline clr/commands.py:313]: Did you mean to commit this testing function?

Comment (mpcusack-color) [reply clr/commands.py:313]: I've added it and removed it enough times I decided to check it in. Might remove later.

**Pull request merged**